crypto/ocsp/ocsp_test.cc

```diff
@@ -766,3 +768,93 @@ TEST_P(OCSPRequestTest, OCSPRequestParse) {
     ASSERT_FALSE(ocspRequest);
   }
 }
+
+static const char good_http_response_hdr[] =
+        "HTTP/1.1 200 OK\r\n"
+        "Content-Type: application/ocsp-response\r\n"
+        "Content-Length: ";
+
+static const char malformed_http_response_hdr[] =
+        "HTTP200 OK\r\n"
+        "Content-Type: application/ocsp-response\r\n"
+        "Content-Length: ";
+
+// Only status code 200 should be accepted.
+static const char not_ok_http_response_hdr[] =
+        "HTTP/1.1 404 OK\r\n"
+        "Content-Type: application/ocsp-response\r\n"
+        "Content-Length: ";
+
+// This should parse. Only the status code is mandatory, subsequent lines are
+// optional.
+static const char no_type_http_response_hdr[] =
+        "HTTP/1.1 200 OK\r\n"
+        "Content-Length: ";
+
+// This should parse. Only the status code is mandatory, additional information
+// is optional.
+static const char no_info_http_response_hdr[] =
```

Comment:
Fuzzing might cover this, but should we add some tests for bad/malformed http responses?